src/modules/flow/console/console.c

```diff
@@ -149,6 +149,22 @@ console_in_process(struct sol_flow_node *node, void *data, uint16_t port, uint16
         }
 
         fprintf(mdata->fp, "} (blob)%s\n", mdata->suffix);
+    } else if (packet_type == SOL_FLOW_PACKET_TYPE_JSON_OBJECT) {
+        struct sol_blob *val;
+
+        int r = sol_flow_packet_get_json_object(packet, &val);
+        SOL_INT_CHECK(r, < 0, r);
+
+        console_output(mdata, "%.*s (JSON object)", (int)val->size,
```

Comment:
could be, but usually we don't print blobs like that, only when we know it's a string such as in JSON.

What we could have is a `sol_str_slice_from_blob()`  helper and then use that macro, one line more, but at least it's cleaner.